src/utils/mod.rs

```diff
@@ -46,6 +46,22 @@ impl PartialEq for Node {
     }
 }
 
+impl Node {
+    pub fn num_siblings(&self) -> u64 {
+        self.children.len() as u64
+    }
+
+    pub fn get_children_from_node(&self, is_reversed: bool) -> impl Iterator<Item = Node> {
+        let children = self.children.clone();
```

Comment:
Do we need the clone of children here? If you think we do - then that's probably what the map was doing in the original method. 

 @baile320